CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift

```diff
@@ -0,0 +1,454 @@
+        for value in barValues {
+            let count = barValues.count
+            let width = valueBarsStackView.frame.width/CGFloat(count)
+            let height = valueBarsStackView.frame.height
+            let valueLabel = UILabel()
+            valueLabel.frame = CGRect(x: 0, y: 0, width: width, height: height)
+            valueLabel.text = "\(value)"
+            valueLabel.textAlignment = .center
+            valueBarsStackView.addArrangedSubview(valueLabel)
+        }
+        layoutSubviews()
+    }
+    
+    private func animateBasedOnCurrentValue(_ value: Double) {
+        if value <= barValues[0] {
+            // MARK: - Case 1: current value is less than the smallest value
+            animateProgressBar(0)
+            return
+        }
+        
+        if let lastBarValue = barValues.last,
+            // MARK: - Case 2: current value is greater or equal than greatest value
+            value >= lastBarValue {
+            
+            markerLabel.centerXAnchor.constraint(equalTo: markerView.centerXAnchor)
+            let trailingMarkerAndLabelConstraint = self.markerView.trailingAnchor.constraint(equalTo: self.markerLabel.trailingAnchor)
+            trailingMarkerAndLabelConstraint.priority = .required
+            animateProgressBar(1.0)
+            return
+        }
+        
+        // MARK: - Case 3: current value falls within the range
+        for segment in 0..<barValues.count - 1 {
+            let start = barValues[segment]
+            let end = barValues[segment+1]
+            let centerConstraint = self.markerLabel.centerXAnchor.constraint(equalTo: self.markerView.centerXAnchor)
+            centerConstraint.priority = .required
+            
+            if value >= start && value < end {
+                animateMarkerWithinRange(from: start, to: end, currentValue: value, offset: Double(segment))
+                break
+            }
+        }
+    }
+    
+    private func animateMarkerWithinRange(from bottomRange: Double, to topRange: Double, currentValue: Double, offset: Double) {
+        
+        let range = topRange - bottomRange
+        let currentRange = currentValue - bottomRange
+        let currentRatio = currentRange / range
+        
+        let finalPercent = currentRatio/Double(barColors.count) + (offset*0.2)
+        animateProgressBar(finalPercent)
+    }
+    
+    private func animateProgressBar(_ percentOfBar: Double) {
+        updateConstraints()
+        let change = (self.colorBarsStackView.frame.width * CGFloat(percentOfBar) + (self.markerView.frame.size.width/(3/2)))
+        if animatable {
+            UIView.animate(withDuration: 1, animations: {
+                self.markerView.center.x += change
+                self.view.layoutIfNeeded()
+            })
+        } else {
+            self.markerView.center.x += change
+            self.view.layoutIfNeeded()
+        }
+    }
+}
+
+extension Reactive where Base: CardPartProgressBarView {
```

Comment:
Awesome! Thanks for keeping bindability in mind!

Reply:
Thank you.